### ISSUE TITLE
Statistics Information

### ISSUE THREAD

**Issue by grahamjenson** [grahamjenson/ger#23]
Query an ESM about its statistics, e.g.

1. Number of total Events
2. Number of total People 
3. Number of total Things

**Comment by thelinuxlich** [comment]
I propose we add Redis as dependency specifically for this. It will be easy and fast to get whatever metrics we want

**Comment by thelinuxlich** [comment]
For reference: http://oldblog.antirez.com/post/take-advantage-of-redis-adding-it-to-your-stack.html

**Comment by grahamjenson** [comment]
I don't want to add redis (or any external application) as an absolute dependency to GER because it locks GER into a particular kind of use. That is the main reason for the different ESMs being available, to provide choice and being able to leverage different technologies. I more want to have the ESMs support some kind of statistics output to allow external applications to provide some kind of dashboard, similar to rethinkdbs one. 

The different ESMs might have different statistics as outputs as some statistics might be impossible/difficult to get from particular ESMS, e.g. a precise count of events in the Postgres ESM is pretty expensive. Having some kind of standard is needed so that all ESM's can have a common interface to test and use.

**Comment by thelinuxlich** [comment]
I see, well, it could be a plugin.

**Comment by grahamjenson** [comment]
Yeah. If you can think of a decent way to have ger with a plugin architecture that would be awesome. Especially since then we can seperate out the ESMs into different modules, and even define different heuristics for them. I have not invested any time into this, but if you have any ideas I have created another issue for that :) https://github.com/grahamjenson/ger/issues/36
